examples/flutter_gallery/lib/demo/text_field_demo.dart

```diff
@@ -79,26 +112,19 @@ class TextFieldDemoState extends State<TextFieldDemo> {
             new Input(
               hintText: 'What do people call you?',
               labelText: 'Name',
-              formField: new FormField<String>(
-                // TODO(mpcomplete): replace with person#name=
-                setter:  (String val) { person.name = val; },
-                validator: _validateName
-              )
```

Comment:
I like your original vision but I don't see how to make it work with external changes to the model. It'd need some sort of scaffolding around it.

I'm not sure where to go with my patch here.